### DISCUSSION TITLE
MariaDB support in HammerDB

### DISCUSSION THREAD

**Discussion by shjafri** [TPC-Council/HammerDB#257] [Q&A] [answered]
Could you please let me know when MariaDB support in HammerDB be available?

For the time being, I am using HammerDB 4.1 on RHEL 8.4 and selecting MySQL TPROC-C and have testdb1 created in MariaDB. But when I launch the hammerdb gui and run build schema after entering params in options it fails with attached error:

![Snip20210701_9](https://user-images.githubusercontent.com/24300199/124115724-234bdf80-da3c-11eb-90cd-01bb5ca1baf2.png)

I have also tried librarycheck from hammerdbcli and MySQL library fails to load:

![Snip20210701_11](https://user-images.githubusercontent.com/24300199/124116218-b258f780-da3c-11eb-88a8-66baf37a9684.png)

MY $LD_LIBRARY_PATH is a follows:

![Snip20210701_12](https://user-images.githubusercontent.com/24300199/124116420-f0561b80-da3c-11eb-8ecd-6fd30d4a53cb.png)

I am stuck at this point. I would really appreciate any help or guidance you can give to fix this and get past this?

Regards,
-Saad

**Comment by sm-shaw** [comment, marked as answer]
Hi,
HammerDB support for MariaDB will be in the next release v4.2 - this is complete and will be available as a binary release very soon. 
Until then HammerDB v4.1 and earlier needs the MySQL 8.0.X client library to connect to MariaDB. There is documentation here on how to do this here https://www.hammerdb.com/docs/ch01s08.html
So you need the file libmysqlclient.so.21 - note that, that is the only file you need and it needs to be in your LIBRARY_PATH as you show above.  If you follow the links from here https://dev.mysql.com/downloads/c-api/  it says C API (libmysqlclient) is included in MySQL 8.0 / Linux: The Client Utilities Package is available from the MySQL Community Server download page. If you download the minimal install (or the full install if you prefer) and untar the package libmysqlclient.so.21 can be found in the lib directory. Wherever you extracted the tar file add this lib directory to your LD_LIBRARY_PATH and use librarycheck (and/or ldd for deeper troubleshooting as described in the documentation). HammerDB will then be able to use the MySQL workload against MariaDB. 
When available with v4.2 the MariaDB client will look for "libmariadb.so.3" so will not need a separate MySQL client installation.
